zegami_sdk/collection.py

```diff
@@ -345,6 +357,40 @@ def download_single(index, url):
             ordered.append(images[i])
         return ordered
 
+    def upload_image(self, path):
+        mime_type = ''
+        file_name = os.path.basename(path)
+        file_ext = os.path.splitext(path)[-1]
+        # Check if the file has a correct extension
+        if file_ext in self.IMAGE_MIMES.keys():
+            mime_type = self.IMAGE_MIMES[file_ext]
```

Comment:
It might be possible to use the mimetypes module for this - although there might be a reason the CLI doesn't do this.

We do definitely the ability for the user to specify the mime type, this is essential for some WSI formats.